File: SRC/BE/user-driver-service/src/main/java/com/trung/userdriverservice/dto/response/UserResponse.java
package com.trung.userdriverservice.dto.response;

import com.trung.userdriverservice.util.enums.ApprovalStatus;
import com.trung.userdriverservice.util.enums.DriverStatus;
import com.trung.userdriverservice.util.enums.Role;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
public class UserResponse {
    private Long id;
    private String phoneNumber;
    private String email;
    private String fullName;
    private Role role;
    private Boolean isLocked;
    private String lockedReason;
    private LocalDateTime lockedAt;
    private DriverStatus status;
    private ApprovalStatus approvalStatus;
    private String rejectionReason;
    private LocalDateTime approvedAt;
    private String vehicleType;
    private String licensePlate;
    private String vehicleModel;
    private LocalDateTime createdAt;
}

File: SRC/APP/app/src/main/java/com/example/omnigo/features/auth/data/remote/dto/response/UserResponse.kt
package com.example.omnigo.features.auth.data.remote.dto.response

data class UserResponse(
    val id: Long = 0L,
    val phoneNumber: String? = null,
    val email: String? = null,
    val fullName: String? = null,
    val role: String? = null,
    val status: String? = null,
    val vehicleType: String? = null,
    val licensePlate: String? = null,
    val vehicleModel: String? = null,
    val createdAt: String? = null,
    val accessToken: String? = null,
    val refreshToken: String? = null
)